src/test/ui/rfc1598-generic-associated-types/parse/in-trait-impl.rs

```diff
@@ -0,0 +1,21 @@
+// Copyright 2012 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// compile-flags: -Z parse-only
+// must-compile-successfully
+
+#![feature(generic_associated_types)]
+
+impl<T> Baz for T where T: Foo {
+    //FIXME(sunjay): This should parse successfully
+    type Quux<'a> = <T as Foo>::Bar<'a, 'static>;
```

Comment:
@nikomatsakis So I marked this as FIXME because I expected that it wouldn't parse, but for some reason it does...Any idea what's going on? We didn't add any parsing for this as far as I know...

Reply:
Note that while this parses, it doesn't work currently past the parsing stage. That's why it's commented out in the run-pass tests above. Question: Is this a problem with the name resolution? In that case we should probably fix it in this PR, right?